docs/context.rst

```diff
@@ -36,21 +36,64 @@ generally constists of something like the following:
 *   Send all given context with any events during the request lifecycle
 *   Cleanup context
 
-There are three primary methods for providing request context:
+There are three primary methods for providing request context.
+
+User Context
+~~~~~~~~~~~~
+
+User context describes the current actor.
 
 .. sourcecode:: ruby
 
     # bind the logged in user
-    Raven.user_context email: 'foo@example.com'
+    Raven.user_context {
+      # a unique ID which represents this user
+      id: 1,
+
+      # the actor's email address, if available
+      email: 'foo@example.com',
+
+      # the actor's username, if available
+      username: 'foo',
+
+      # the actor's IP address, if available
+      ip_address: '127.0.0.1'
+    }
+
+When dealing with anonymous users you will still want to send basic user context to ensure Sentry can count them against the unique users:
+
+    Raven.user_context {
+      # the actor's IP address
+      ip_address: '127.0.0.1'
+    }
+
+Tags
+~~~~
+
+You can provide a set of key/value pairs called tags which Sentry will index and aggregate. This will help you understand the distribution of issues, as well as enabling easy lookup via search.
+
+.. sourcecode:: ruby
 
     # tag the request with something interesting
-    Raven.tags_context interesting: 'yes'
+    Raven.tags_context {
+      language: 'enUS',
+      timezone: 'PST'
+    }
```

Comment:
This isn't valid Ruby, because the curly braces get interpreted as a block. This is valid though:
```ruby
Raven.tags_context language: 'enUS', timezone: 'PST'